rsconf/component/__init__.py

```diff
@@ -451,6 +458,11 @@ def _assert_host_path(host_path):
     )
 
 
+def _bash_quote(value):
+    v = str(value).replace("\\", r"\\").replace("'", r"\'")
+    return f"$'{v}'"
```

Comment:
Neat. For anyone else or my future self the `$''` syntax is called ansi-c quoting https://www.gnu.org/software/bash/manual/html_node/ANSI_002dC-Quoting.html